### PR TITLE
CIRC-1578 - Add the field "requester.patronGroup" to the Requester Object in the Staff Slip context

### PR DESCRIPTION
**Pull request by gurleenkaurbp**
## Purpose
Add the field "requester.patronGroup" to the Requester Object in the Staff Slip context
https://issues.folio.org/browse/CIRC-1578
## Approach
Determined the Patron Group via the patron group ID and set it in the requester's context json. 

## Learning
Async program handling of completableFuture result that is returned to set the value in patronGroup.

##Code coverage on the new lines 
image

##Evidence 
{"pickSlips":[{"item":{"title":"The Long Way to a Small, Angry Planet","primaryContributor":"Chambers, Becky","allContributors":"Chambers, Becky","barcode":"036000291452","status":"Paged","enumeration":null,"volume":null,"chronology":null,"yearCaption":"","materialType":"Book","loanType":"Can Circulate","copy":"","numberOfPieces":null,"descriptionOfPieces":null,"effectiveLocationSpecific":"3rd Floor","effectiveLocationLibrary":"Djanogly Learning Resource Centre","effectiveLocationCampus":"Jubilee Campus","effectiveLocationInstitution":"Nottingham University","callNumber":"123456","callNumberPrefix":"PREFIX","callNumberSuffix":"SUFFIX"},"request":{"requestID":"6754b0cc-6d87-4016-bfa3-7d4e9173f5cf","servicePointPickup":"Circ Desk 1"},"requester":{"firstName":"James","preferredFirstName":null,"lastName":"Rodwell","middleName":null,"barcode":"6430530304",**_"patronGroup":"Regular Group"_**}}],"totalRecords":1}

{"item":{"title":"The Long Way to a Small, Angry Planet","primaryContributor":"Chambers, Becky","allContributors":"Chambers, Becky","barcode":"036000291452","status":"Awaiting pickup","enumeration":null,"volume":null,"chronology":null,"yearCaption":"","materialType":"Book","loanType":"Can Circulate","copy":"","numberOfPieces":null,"descriptionOfPieces":null,"effectiveLocationSpecific":"3rd Floor","effectiveLocationLibrary":"Djanogly Learning Resource Centre","effectiveLocationCampus":"Jubilee Campus","effectiveLocationInstitution":"Nottingham University","callNumber":"123456","callNumberPrefix":"PREFIX","callNumberSuffix":"SUFFIX","lastCheckedInDateTime":"2023-01-18T12:33:19.999Z","fromServicePoint":"Circ Desk 1"},"request":{"requestID":"233da317-f394-4f3d-9704-c9801b47a90f","servicePointPickup":"Circ Desk 1","requestExpirationDate":"2019-07-30T23:59:59.000Z","holdShelfExpirationDate":"2019-08-31T00:00:00.000Z","deliveryAddressType":"Home","patronComments":"I need the book"},"requester":{"firstName":"Steven","preferredFirstName":"Jon","lastName":"Jones","middleName":"Jacob","barcode":"5694596854","addressLine1":"12 Grimmauld Place","addressLine2":null,"city":"London","region":"London region","postalCode":"123456","countryId":"UK",**_"patronGroup":"Regular Group"_**}}

CheckIn By Barcode - 
image
image

Print Staff slips for service point - 
image
image

### CODE REVIEW

File: src/main/java/org/folio/circulation/resources/PickSlipsResource.java [hunk 1]
```diff
@@ -9,18 +9,15 @@
...
+
 import static org.folio.circulation.support.results.Result.succeeded;
 import static org.folio.circulation.support.results.ResultBinding.flatMapResult;
 
-import java.util.Collection;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-import java.util.UUID;
+import java.util.*;
```
Comment (Arghya22) [inline src/main/java/org/folio/circulation/resources/PickSlipsResource.java:16]: Please remove wild card import

File: src/main/java/org/folio/circulation/resources/PickSlipsResource.java [hunk 2]
```diff
@@ -84,7 +84,7 @@ private void getMany(RoutingContext routingContext) {
     final var itemRepository = new ItemRepository(clients);
     final AddressTypeRepository addressTypeRepository = new AddressTypeRepository(clients);
     final ServicePointRepository servicePointRepository = new ServicePointRepository(clients);
-
+    PatronGroupRepository patronGroupRepository = new PatronGroupRepository(clients);
```
Comment (Arghya22) [inline src/main/java/org/folio/circulation/resources/PickSlipsResource.java:87]: Can be a final field

File: src/test/java/api/loans/CheckInByBarcodeTests.java [hunk 3]
```diff
@@ -53,8 +53,7 @@
 import static org.hamcrest.Matchers.hasSize;
 import static org.hamcrest.core.Is.is;
 import static org.hamcrest.core.IsNot.not;
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.fail;
+import static org.junit.jupiter.api.Assertions.*;
```
Comment (Arghya22) [inline src/test/java/api/loans/CheckInByBarcodeTests.java:56]: wildcard

Comment (gurleenkaurbp) [reply src/test/java/api/loans/CheckInByBarcodeTests.java]: done

File: src/main/java/org/folio/circulation/resources/PickSlipsResource.java (hunk 2 above)
Comment (gurleenkaurbp) [reply src/main/java/org/folio/circulation/resources/PickSlipsResource.java]: done

File: src/main/java/org/folio/circulation/resources/PickSlipsResource.java (hunk 1 above)
Comment (gurleenkaurbp) [reply src/main/java/org/folio/circulation/resources/PickSlipsResource.java]: Done

**Review by arin-epm — APPROVED** [review]

**Review by Arghya22 — APPROVED** [review]

**Pull request merged**